modules/objdetect/perf/perf_qrcode_pipeline.cpp

```diff
@@ -45,75 +48,72 @@ PERF_TEST_P_(Perf_Objdetect_QRCode, decode)
-}
+static std::set<std::pair<std::string, QRDetectMethod>> dropped_samples = {{"5_qrcodes.png", QRDetectMethod::ARUCO}};
 
 PERF_TEST_P_(Perf_Objdetect_QRCode_Multi, detectMulti)
 {
-    const std::string name_current_image = GetParam();
+    const std::string name_current_image = get<0>(GetParam());
+    const QRDetectMethod method = get<1>(GetParam());
     const std::string root = "cv/qrcode/multiple/";
 
     std::string image_path = findDataFile(root + name_current_image);
     Mat src = imread(image_path);
     ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
-    std::vector<Point2f> corners;
-    QRCodeDetector qrcode;
+    QRDetectorParameters params;
+    params.qrDetectMethod = method;
+    std::vector<Point> corners;
+    QRCodeDetector qrcode(params);
     TEST_CYCLE() ASSERT_TRUE(qrcode.detectMulti(src, corners));
-    sort(corners.begin(), corners.end(), compareCorners);
-    SANITY_CHECK(corners);
-}
-
-static inline bool compareQR(const pair<string, Mat>& v1, const pair<string, Mat>& v2) {
-    return v1.first < v2.first;
+    const int pixels_error = 7;
+    check_qr(root, name_current_image, "multiple_images", corners, {}, pixels_error, true);
+    SANITY_CHECK_NOTHING();
 }
 
 #ifdef HAVE_QUIRC
 PERF_TEST_P_(Perf_Objdetect_QRCode_Multi, decodeMulti)
 {
-    const std::string name_current_image = GetParam();
+    const std::string name_current_image = get<0>(GetParam());
+    const QRDetectMethod method = get<1>(GetParam());
     const std::string root = "cv/qrcode/multiple/";
     std::string image_path = findDataFile(root + name_current_image);
     Mat src = imread(image_path);
     ASSERT_FALSE(src.empty()) << "Can't read image: " << image_path;
-    QRCodeDetector qrcode;
+    QRDetectorParameters params;
+    params.qrDetectMethod = method;
+    QRCodeDetector qrcode(params);
     std::vector<Point2f> corners;
     ASSERT_TRUE(qrcode.detectMulti(src, corners));
     std::vector<Mat> straight_barcode;
     std::vector< cv::String > decoded_info;
     TEST_CYCLE()
     {
         ASSERT_TRUE(qrcode.decodeMulti(src, corners, decoded_info, straight_barcode));
-        for(size_t i = 0; i < decoded_info.size(); i++)
-        {
-            ASSERT_FALSE(decoded_info[i].empty());
-        }
+    }
+    if (dropped_samples.find({name_current_image, method}) != dropped_samples.end()) {
+        SANITY_CHECK_NOTHING();
+        return;
+    }
+    ASSERT_TRUE(decoded_info.size() > 0ull);
+    for(size_t i = 0; i < decoded_info.size(); i++) {
+        ASSERT_FALSE(decoded_info[i].empty());
     }
     ASSERT_EQ(decoded_info.size(), straight_barcode.size());
-    vector<pair<string, Mat> > result;
-    for (size_t i = 0ull;  i < decoded_info.size(); i++) {
-        result.push_back(make_pair(decoded_info[i], straight_barcode[i]));
+    vector<Point> corners_result;
+    for (const Point2f& corner : corners) {
+        corners_result.push_back(corner);
     }
```

Comment:
corners_result size is known. May be pre-allocated.

Reply:
fixed
```
    vector<Point> corners_result(corners.size());
    for (size_t i = 0ull; i < corners_result.size(); i++) {
        corners_result[i] = corners[i];
    }
```